===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_constant_reverse:lookups)) at 36 (line 689, column 10),
        Insert: if ({ p11_constant_reverse:n$31 == { 0 } }) free(*(p11_constant_reverse:lookups->buckets)) at 35 (line 689, column 10) }
===== Actual patch in diff format =====
@@ -687,6 +687,8 @@ p11_constant_reverse (bool nick)
 				}
 			} else {
 				if (!p11_dict_set (lookups, (void *)table[j].name, (void *)&table[j].value))
+					free(lookups);
+					free(lookups->buckets);
 					return_val_if_reached (NULL);
 			}
 		}
